### PR TITLE
Fix max_train_samples, stratified dataset split

### PR DESCRIPTION
**Pull request by natuan**
Fix max_train_samples, max_eval_samples for train/val split from using validation_ratio.

Fix stratified split (used currently for IMDB) when the datasets package does not support this feature. Keep the original code for reproducibility of shipped IMDB models.

This is to resolve the ticket: https://app.asana.com/0/1201652400520766/1202932644760687/f

### CODE REVIEW

File: src/sparseml/transformers/text_classification.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import numpy as np
 import transformers
 from datasets import load_dataset, load_metric
+from sklearn.model_selection import StratifiedShuffleSplit
```
Comment (rahul-tuli) [inline src/sparseml/transformers/text_classification.py:38]: Is `sklearn` installed as a part of our transformers fork? if not could we add it to `setup.py`

File: src/sparseml/transformers/text_classification.py [hunk 2]
```diff
@@ -674,19 +667,26 @@ def _create_train_dataset(raw_datasets, max_train_samples):
...
-                )
+            train_dataset = (
+                raw_datasets["train"] if train_dataset is None else train_dataset
+            )
             train_dataset, eval_dataset = _split_train_val(
                 train_dataset, data_args.validation_ratio
             )
+            if (
+                data_args.max_train_samples is not None
+                and len(train_dataset) > data_args.max_train_samples
```
Comment (rahul-tuli) [inline src/sparseml/transformers/text_classification.py:678]: Should we also support cases where `len(train_dataset) < data_args.max_train_samples` ?

Comment (rahul-tuli) [reply src/sparseml/transformers/text_classification.py:678]: Or would that qualify as a usage error

Comment (natuan) [reply src/sparseml/transformers/text_classification.py:678]: No (if max_train_samples >= len of the dataset then we just keep using it)

File: src/sparseml/transformers/text_classification.py (hunk 1 above)
Comment (natuan) [reply src/sparseml/transformers/text_classification.py:38]: Added scikit-learn to setup.py

**Review by rahul-tuli — APPROVED** [review]

**Review by anmarques — APPROVED** [review]

**Pull request merged**